scripts/fulfillURLManifest.py

```diff
@@ -0,0 +1,110 @@
+import boto3
+import json
+import os
+import logging
+from botocore.exceptions import ClientError
+from model import Link
+from managers import DBManager, s3
+
+s3_client = boto3.client("s3")
+
+bucketName = 'drb-files-qa'
+
+# class fulfillURLManifest():
+
+#     def __init__(self):
+
+#         # Connect to database
+#         self.generateEngine()
+#         self.createSession()
+
+#         # S3 Configuration
+#         self.s3Bucket = bucketName
+#         self.createS3Client()
+
+def main():
+
+    '''Replacing pdf/epub links in with fulfill urls in manifest JSON files'''
+
+    batches = load_batch()
+    for batch in batches:
+        for c in batch['Contents']:
+            currKey = c['Key']
+            metadataObject = s3_client.get_object(Bucket= bucketName, Key= f'{currKey}')
+            update_batch(metadataObject, bucketName, currKey)
+    
+
+def update_batch(metadataObject, bucketName, currKey):
+
+    dbManager = DBManager(
+        user=os.environ.get('POSTGRES_USER', None),
+        pswd=os.environ.get('POSTGRES_PSWD', None),
+        host=os.environ.get('POSTGRES_HOST', None),
+        port=os.environ.get('POSTGRES_PORT', None),
+        db=os.environ.get('POSTGRES_NAME', None)
+    )
+
+    dbManager.generateEngine()
+
+    dbManager.createSession()
+
+    metadataJSON = json.loads(metadataObject['Body'].read().decode("utf-8"))
+    
+    metadataJSON = linkFulfill(metadataJSON, dbManager)
+    metadataJSON = readingOrderFulfill(metadataJSON, dbManager)
+    metadataJSON = resourceFulfill(metadataJSON, dbManager)
+    metadataJSON = tocFulfill(metadataJSON, dbManager)
+
+    fulfillManifest = json.dumps(metadataJSON, ensure_ascii = False, indent = 6)
+
+    try:
+        return s3_client.put_object(Bucket=bucketName, Key=currKey, \
```

Comment:
I agree with your assessment Lyndsey. I'll work on adding putObjectInBucket as the method instead since I also thought it's better for error handling and future proofing the script.